src/Uno.Toolkit.UI/Controls/TabBar/TabBarSelectionIndicatorPresenter.cs

```diff
@@ -323,6 +323,52 @@ private void StopStoryboards()
 			_verticalStoryboard?.Stop();
 		}
 
+		private double TranslateOffset
+		{
+			get
+			{
+				if (GetSelectionIndicator() is not { RenderTransform: CompositeTransform transform }) return 0;
+				{
+					return Owner?.Orientation == Orientation.Horizontal ? transform.TranslateX : transform.TranslateY;
+				}
+			}
+			set
+			{
+				if (GetSelectionIndicator() is not { RenderTransform: CompositeTransform transform }) return;
+				if (Owner?.Orientation == Orientation.Horizontal)
+				{
+					transform.TranslateX = value;
+				}
+				else
+				{
+					transform.TranslateY = value;
+				}
+			}
+		}
+
+		private void StopRunningAnimation()
+		{
+			var storyboard = GetStoryboardForCurrentOrientation();
+			if (storyboard != null && storyboard.GetCurrentState() != ClockState.Stopped)
+			{
+				storyboard.Pause();
+				var currentOffset = TranslateOffset;
+				storyboard.Stop();
+				TranslateOffset = currentOffset;
```

Comment:
some comment to explain the why would be helpful here
something to snapshot animated value~